Library/Homebrew/cmd/config.rb

```diff
@@ -119,9 +119,15 @@ def describe_system_ruby
     s
   end
 
-  def java_version
-    java = `java -version 2>&1`.lines.first.chomp
-    java =~ /java version "(.+?)"/ ? $1 : java
+  def describe_java
+    if which("java").nil?
+      "N/A"
+    elsif OS.mac? && `/usr/libexec/java_home --failfast` && !$?.success?
```

Comment:
Also, probably don't want to use backticks but instead use `system` and redirect `2>&1`.